### PR TITLE
test(bridge): Add integration tests for transfer panel

### PR DESCRIPTION
**Pull request by chrstph-dvx**

### CODE REVIEW

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.integration.helpers.tsx
Comment (fionnachan) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.integration.helpers.tsx:1]: formatting failure on L434

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TokenRow.tsx [hunk 1]
```diff
@@ -98,10 +98,7 @@ function TokenListInfo({ token }: { token: ERC20BridgeToken | null }) {
   }
 
   if (addressesEqual(token.address, constants.AddressZero)) {
-    if (
-      (sourceChainNativeCurrency.isCustom && destinationChainNativeCurrency.isCustom) ||
-      networks.sourceChain.id === ChainId.ApeChain
-    ) {
+    if (sourceChainNativeCurrency.isCustom && destinationChainNativeCurrency.isCustom) {
```
Comment (fionnachan) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TokenRow.tsx:101]: what's this change for?

File: packages/arb-token-bridge-ui/src/components/TransferPanel/DestinationTokenButton.tsx [hunk 2]
```diff
@@ -30,8 +29,7 @@ export function DestinationTokenButton({
 
   const [dialogProps, openDialog] = useDialog2();
 
-  const { childChainProvider } = useNetworksRelationship(networks);
-  const nativeCurrency = useNativeCurrency({ provider: childChainProvider });
+  const nativeCurrency = useNativeCurrency({ provider: networks.destinationChainProvider });
```
[the remark this answers is not in the document]
Comment (fionnachan) [reply packages/arb-token-bridge-ui/src/components/TransferPanel/DestinationTokenButton.tsx]: this is wrong. it introduces a regression bug for custom native token chain withdrawals.

Image

**Review by fionnachan — COMMENTED** [review]
this PR shouldn't change any logic in the code because a seemingly small change could break the app

let's only introduce the tests without making any changes unless they are purely HTML element changes

File: yarn.lock [hunk 3]
```diff
@@ -13920,12 +13943,11 @@ tar-stream@^2.1.4:
     readable-stream "^3.1.1"
 
 tar-stream@^3.1.5:
-  version "3.1.8"
-  resolved "https://registry.yarnpkg.com/tar-stream/-/tar-stream-3.1.8.tgz#a26f5b26c34dfd4936a4f8a9e694a8f5102af13d"
-  integrity sha512-U6QpVRyCGHva435KoNWy9PRoi2IFYCgtEhq9nmrPPpbRacPs9IH4aJ3gbrFC8dPcXvdSZ4XXfXT5Fshbp2MtlQ==
+  version "3.1.7"
```
Comment (fionnachan) [inline yarn.lock]: version regression

File: packages/arb-token-bridge-ui/src/components/App/useArbTokenBridgeBootstrap.ts
Comment (fionnachan) [inline packages/arb-token-bridge-ui/src/components/App/useArbTokenBridgeBootstrap.ts:1]: can you explain why this is essential?

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TokenRow.tsx (hunk 1 above)
Comment (chrstph-dvx) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TokenRow.tsx:101]: ApeChain to Arb1, shows ETH (native currency of destination chain) as "Native currency on ArbitrumOne" and not "LiFi tokens"

File: packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasEstimates.ts [hunk 4]
```diff
@@ -167,6 +167,10 @@ export function useGasEstimates({
 
   const { data: gasEstimates, error } = useSWR(
     () => {
+      if (amountToTransfer.lte(0)) {
+        return null;
+      }
```
Comment (chrstph-dvx) [inline packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasEstimates.ts]: For 0 amount, we're still trying to estimate gas, which throw RPC errors. During tests, it shows as error:
```bash
 Test Files  4 passed (4)
      Tests  31 passed (31)
     Errors  7 errors
```

File: packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+function getRedirectedUrl() {
+  const [redirectTarget] = redirectMock.mock.calls.at(-1) ?? [];
+
+  if (!redirectTarget) {
+    throw new Error('Expected redirect to be called.');
+  }
+
+  return new URL(redirectTarget, 'https://example.com');
```
Comment (dewanshparashar) [inline packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts]: Why do we have example.com as the fallback here?

File: packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+vi.mock('@/bridge/util/networks', async (importActual) => {
+  const actual = await importActual<typeof import('@/bridge/util/networks')>();
+
+  return {
+    ...actual,
+    registerLocalNetwork: registerLocalNetworkMock,
+  };
+});
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
```
Comment (dewanshparashar) [inline packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts]: NIT: Can't this be reused from constants?

File: packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  return new URL(redirectTarget, 'https://example.com');
+}
+
+describe('initializeBridgePage sanitization', () => {
+  beforeEach(() => {
+    redirectMock.mockClear();
+    registerLocalNetworkMock.mockClear();
+  });
+
+  it('sets destinationToken to zero address for first-load apechain -> superposition', async () => {
```
Comment (dewanshparashar) [inline packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts:129]: Suggestion:  ApeChain->SuperPosition is a good corner case, We should also consider addding tests for: 
1. default state 
2. invalid source chain network 
3. invalid destination chain network, 
4. invalid both networks
5. both networks valid.

Should be pretty low lift but would considerably increase coverage.

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.usdc.integration.test.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  setSourceToken,
+  setupTransferPanelLifiIntegrationSuite,
+  usdcTokenByChain,
+} from './TransferPanel.integration.helpers';
+
+const usdcCases: RouteTokenCase[] = [
+  {
+    sourceChain: 'ethereum',
+    destinationChain: 'apechain',
+    expectedSourceToken: usdcTokenByChain.ethereum,
```
Comment (dewanshparashar) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.usdc.integration.test.tsx]: Hmm.. I would just call it `sourceToken` and `destinationToken`, since the `expected` action is already covered by the test helper name - `expectTokenButtonToken`. Otherwise reading these route cases initially felt like if sourceChain/destinationChain are ethereum+Apechain, then the expected tokens should be ETH and APE.. which is confusing.

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.usdc.integration.test.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    async ({ sourceChain, destinationChain, expectedSourceToken, expectedDestinationToken }) => {
+      await renderTransferPanel({
+        sourceChain,
+        destinationChain,
+      });
+
+      await setSourceToken(expectedSourceToken);
+      await setDestinationToken(expectedDestinationToken);
+
+      await expectTokenButtonToken({
```
Comment (dewanshparashar) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.usdc.integration.test.tsx]: `expectTokenButtonToken` -> `expectTokenButtonContent({isDestination: boolean, tokenExpectation: source/destintionToken })`

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.swap.integration.test.tsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+const swapCases: RouteTokenCase[] = [
+  {
+    sourceChain: 'ethereum',
+    destinationChain: 'apechain',
+    expectedSourceToken: usdcTokenByChain.ethereum,
+    expectedDestinationToken: wethTokenByChain.apeChain,
+  },
+  {
+    sourceChain: 'apechain',
+    destinationChain: 'ethereum',
```
Comment (dewanshparashar) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.swap.integration.test.tsx:23]: In general, why are we re-defining the data structure for token definition from chain-first present in the whole repo ie. `CommonAddress.Ethereum.USDC` to token-first here - ie, `usdcTokenByChain.ethereum`? I think we should stick to the `CommonAddress.chainName.token` convention and reuse it here.

File: packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.integration.helpers.tsx [hunk 11]
```diff
@@ -0,0 +1,639 @@
...
+
+export const APE_TOKEN_LOGO = '/images/ApeTokenLogo.svg';
+export const WETH_TOKEN_LOGO =
+  'https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2/logo.png';
+export const USDC_TOKEN_LOGO =
+  'https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/arbitrum/assets/0xaf88d065e77c8cC2239327C5EDb3A432268e5831/logo.png';
+export const USDT_TOKEN_LOGO =
+  'https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0xdAC17F958D2ee523a2206206994597C13D831ec7/logo.png';
+export const WETH_SUPERPOSITION_ROW_LOGO =
+  'https://static.debank.com/image/eth_token/logo_url/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2/61844453e63cf81301f845d7864236f6.png';
```
Comment (dewanshparashar) [inline packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.integration.helpers.tsx]: Some of these logos are redefined multiple times in the codebase. Ideally they should be inferred from a common constant file here + in the codebase.

File: packages/arb-token-bridge-ui/vitest.integration.config.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    snapshotFormat: {
+      escapeString: true,
+    },
+    testTimeout: 120_000,
+    include: ['./src/**/*.integration.test.ts', './src/**/*.integration.test.tsx'],
+    env: loadEnv('', '../app/', ''),
+    environment: 'happy-dom',
+    environmentOptions: {
+      happyDOM: {
+        url: 'http://localhost:3000',
```
Comment (fionnachan) [inline packages/arb-token-bridge-ui/vitest.integration.config.ts:22]: wouldn't work if another port is used? i guess that's ok
is there an option to override this on cli?

File: packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts (hunk 6 above)
Comment (chrstph-dvx) [reply packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts]: good point, replaced with constants from ether

File: packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts (hunk 5 above)
Comment (chrstph-dvx) [reply packages/app/src/utils/__tests__/sanitizeAndRedirect.test.ts]: It's just a placeholder, but I changed it to portal.arbitrum.io

File: package.json [hunk 13]
```diff
@@ -10,7 +10,9 @@
     "build": "pnpm --filter app build",
     "start": "pnpm --filter app start",
     "audit:ci": "audit-ci --config ./audit-ci.jsonc",
-    "test:ci": "pnpm --filter arb-token-bridge-ui test:ci",
+    "test:unit": "pnpm --filter arb-token-bridge-ui test:unit",
```
Comment (fionnachan) [inline package.json:13]: you forgot to change `test:ci` to `test:unit` on the `package.json` in `/arb-token-bridge-ui` so now the unit tests are skipped

Image

File: packages/arb-token-bridge-ui/src/components/App/useArbTokenBridgeBootstrap.ts
Comment (chrstph-dvx) [reply packages/arb-token-bridge-ui/src/components/App/useArbTokenBridgeBootstrap.ts:1]: Otherwise token list are not properly populated during integration tests, and I think having them populated is better than mocking them

**Review by dewanshparashar — DISMISSED** [review]

**Review by dewanshparashar — APPROVED** [review]

**Review by fionnachan — APPROVED** [review]
